tests/src/core/testqgsexpression.cpp

```diff
@@ -1718,7 +1718,11 @@ class TestQgsExpression: public QObject
       QTest::newRow( "year with interval" ) << "year(tointerval('2 years'))" << false << QVariant( 2.0 );
       QTest::newRow( "age" ) << "age('2012-06-30','2012-06-28')" << false << QVariant::fromValue( QgsInterval( 172800 ) );
       QTest::newRow( "negative age" ) << "age('2012-06-28','2012-06-30')" << false << QVariant::fromValue( QgsInterval( -172800 ) );
+#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
       QTest::newRow( "big age" ) << "age('2000-01-01','1000-01-01')" << false << QVariant::fromValue( QgsInterval( 31556908800LL ) );
+#else
```

Comment:
Do you know why the result is different with Qt6 ? A quick test with Python3 returns the same result as Qt5

```python3
> (datetime.strptime('2000-01-01', '%Y-%m-%d') - datetime.strptime('1000-01-01', '%Y-%m-%d')).total_seconds()
31556908800.0
```

Reply:
I don't, but that's a BIG interval, so I'm not that surprised if it encompasses some ranges where leap second calculations are involved...

Reply:
The [documentation](https://doc.qt.io/qt-5/qdatetime.html#details) says that QDateTime doesn't account for leap seconds.

I tested it with PostGres, and I also get the Qt5 result. I'm wondering if this is a Qt issue.

```sql
select extract(epoch from ('2000-01-01 00:00:00'::timestamp - '1000-01-01 00:00:00'::timestamp));
      extract       
--------------------
 31556908800.000000
```
Anyway, this is non blocking. LGTM

Reply:
The `age` function uses `QDateTime::secsTo()` to calculate the `QgsInterval` in seconds.

Reply:
Hmm, I'll drop this change for now until we can further investigate...

Reply:
I wonder if it's related to this qt 6.2 change: "QDateTime now takes account of time-zones throughout time_t's range (in so far as the platform supplies such information), rather than artificially limiting the range to 1970 - 2037"

Reply:
The difference is 728 seconds = 7 minutes and 8 seconds... If it was due to time-zones, then I would expect it to be a whole number of hours.

Reply:
I've used PyQt6 Qt 6.3.1 and PyQt5 Qt 5.15.3 on Ubuntu 22.04 to test what's going wrong.

The `age()` function uses `QDateTime::secsTo()` which in turn calculates the number of seconds between two dates essentially calculating the difference between the corresponding Epoch timestamps values (with `QDateTime::toMSecsSinceEpoch()/1000`).

The different results of `age('2000-01-01','1000-01-01')` in Qt5 and Qt6 seems due to the fact that the Epoch value of '1000-01-01' is different between the two Qt version (using `toSecsSinceEpoch()`) if no timezone is specified:

Qt5
```Python
>>> QDateTime(QDate(1000,1,1),QTime(0,0,0)).toSecsSinceEpoch()
-30610227600
```
Qt6
```Python
>>> QDateTime(QDate(1000,1,1),QTime(0,0,0)).toSecsSinceEpoch()
-30610226996
```

while the Epoch value of '2000-01-01' is equal in both Qt version:

Qt5 and Qt6
```Python
>>> QDateTime(QDate(2000,1,1),QTime(0,0,0)).toSecsSinceEpoch()
946681200
```

Thus, on my system:
- Qt5: `age('2000-01-01','1000-01-01')` = 946681200 - (-30610227600) = 31556908800 = 365242 days
- Qt6: `age('2000-01-01','1000-01-01')` = 946681200 - (-30610226996) = 31556908196 = 365241 days, 23 hours,  49 minutes, 56 seconds

with a difference of -604 seconds = - 10 minutes and 4 seconds.

It seems the difference in the Epoch value calculated by Qt5 and Qt6 occurs for dates before November 1, 1893, so each age() calculation between one date before and one after such date will return different values in Qt5 and Qt6.

The same Epoch values of Qt6 for naive dates before November 1, 1893 are returned by `timestamp()` function of the Python `datetime` module:

```Python
>>> datetime(1000,1,1,0,0,0).timestamp()
-30610226996.0
>>> datetime(2000,1,1,0,0,0).timestamp()
946681200.0
>>> datetime(2000,1,1,0,0,0).timestamp() - datetime(1000,1,1,0,0,0).timestamp()
31556908196.0
```

anyway the difference between the two datetimes is calculated by the `datetime` module with the same value as Qt5 with:
```Python
>>> (datetime(2000,1,1,0,0,0) - datetime(1000,1,1,0,0,0)).total_seconds()
31556908800.0
```

The behaviour of Qt5 and Qt6 is the same when a timezone is specified:
Qt6
```Python
>>> QDateTime(QDate(1000,1,1),QTime(0,0,0),Qt.TimeSpec.UTC).toSecsSinceEpoch()
-30610224000
>>> QDateTime(QDate(2000,1,1),QTime(0,0,0),Qt.TimeSpec.UTC).toSecsSinceEpoch()
946684800
```

Qt5
```Python
>>> QDateTime(QDate(1000,1,1),QTime(0,0,0),Qt.UTC).toSecsSinceEpoch()
-30610224000
>>> QDateTime(QDate(2000,1,1),QTime(0,0,0),Qt.UTC).toSecsSinceEpoch()
946684800
```

also datetime.timestamp() returns the same values:
```Python
>>> datetime(1000,1,1,0,0,0, tzinfo=timezone.utc).timestamp()
-30610224000.0
>>> datetime(2000,1,1,0,0,0, tzinfo=timezone.utc).timestamp()
946684800.0
```